docs/contributing/999-rfcs/2022-12-08-wing-inputs.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "#2726 Wing Inputs"
+description: "How to create, share, manage inputs"
+---
+
+# Wing Inputs
+- **Author(s)**: @Hasanaburayyan
+- **Submission Date**: {2023-12-08}
+- **Stage**: Draft
+
+## Background
+
+An input is a value used by either an application or platform that is not necessarily defined within the application's source code.
+
+Inputs can be Platform agnostic, such as an a retry count that is dictated by business requirements. Other inputs may be Platform specific, like a DNS configurations.
+
+### Context
+
+There are a few things that are considered out-of-scope for this RFC. 
+- Inputs are all treated as string data
+- Inputs cannot be created, or updated inflight
+
+## Use cases
+
+The following table contains some use cases and is by no means exhaustive. 
+
+| **Use case** | **Secure** | Consumed |
+| ---- | ---- | ---- |
+| Connection strings | Yes | Inflight |
+| SSH Keys | Yes | Inflight |
+| Bucket count | No | Both |
+| Connection strings | Yes | Inflight |
+| SSH Keys | Yes | Inflight |
+| Bucket count | No | Both |
+| Retry counts | No | Inflight |
+| API Keys | Yes | Inflight |
+| DNS information (certificates, hosted zones) | No | Preflight |
+| Autoscaling parameters | No | Preflight |
+| Feature Flags | No | Both |
+| Platform Input (state-file name, etc..) | No | Preflight |
+
+
+## Configuring and Managing Inputs
+
+The following Wing application contains several inputs, in this section of the RFC we will see how inputs can be managed
+
+(draft sketch)
+```js
+let enableNotifications = cloud.Input("/features/notifications");
+let emailList = cloud.Input("/email-list"); // comma seperated string
+
+let inbox = new cloud.Bucket() as "inbox";
+
+// conditionally provision infrastructure based on enableNotification
+if (enableNotifications.asBool()) {
+	inbox.onCreate(inflight () => {
+		// use emailList input inflight
+		for email in emailList.asStr().split(",") {
+			// ...
+		}
+	});
+}
+```
+
+In order for the above code to run there are 2 required inputs `enableNotifications` and `emailList` either one of these inputs can be used inflight or preflight. 
+
+To manage these inputs we will create a `inputs.json` file within our project's folder.
+```json
+{
+	"features": {
+		"notifications": "true"
+	},
+	"email-list": "one@email.com,two@email.com"
+}
+```
```

Comment:
> I wonder if inputs.json can be a replacement for the current [values.yaml](https://www.winglang.io/docs/standard-library/cloud/domain#aws-tf-aws-and-awscdk) file used by some resources like cloud.Domain?

Yea I believe that this will replace the need for `values.yaml` entirely